thanos/charts/templates/query/deployment.yaml

```
@@ -53,18 +53,8 @@ spec:
         {{- range .Values.thanos.query.additionalArgs }}
         - {{ . }}
         {{- end }}
-        {{/* Technically it is possible to add stores and have the default stores all together.*/}}
-        {{/* Enabling standalone would remove the default stores for an empty store list to be maintained entirely manual.*/}}
-        {{- if and (empty .Values.thanos.query.stores) (.Values.thanos.query.standalone) }}
-        {{- fail "WARNING Thanos Query will have not data. Either unset thanos.query.standalone or specify thanos.query.stores." -}}
-        {{- end }}
-        {{- range .Values.thanos.query.stores }}
-        - --store={{ . }}
-        {{- end }}
-        {{- if not .Values.thanos.query.standalone }}
-        - --store=prometheus-operated:10901
-        - --store={{ include "release.name" . }}-store:{{ default "10901" .Values.thanos.grpcAddress }}
-        {{- end }}
+        # --store.sd-files will be deprecated in next release. Correct flag will be --endpoint.sd-config-file
```

Why not using `--endpoint.sd-config-file` straight?

---

i tried it and don't think it's implemented yet in current version

---

I can see it in the docs: https://thanos.io/tip/components/query.md/#:~:text=%2D%2Dendpoint.sd%2Dconfig%2Dfile

---

It should work according to this PR: https://github.com/thanos-io/thanos/pull/7890. Could you please create a follow up issue to keep this in mind?

---

```suggestion
        # --store.sd-files will be deprecated in next release. Correct flag will be --endpoint.sd-config-file (available with v0.38)
```

it is available in v0.38

currently we have [v0.38.0-rc.0](https://github.com/thanos-io/thanos/releases/tag/v0.38.0-rc.0) as a prerelease, as soon as this is in, we can switch.

---

you are looking at the tip
that's our docs https://thanos.io/v0.37/components/query.md/#flags

---

Thanks. You are right.